scripts/utils/generate_shared_addon_xliff.py

```diff
@@ -0,0 +1,76 @@
+#! /usr/bin/env python3
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# This script is used when importing strings upstream into the i18n repo to convert the addons/strings.yaml file into an xliff.
+
+import argparse
+import os
+import sys
+import tempfile
+
+from generate_strings import parseYAMLTranslationStrings
+
+# hack to be able to access shared.py
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '../')))
+from shared import write_en_language, qtquery
+
+
+# parseYAMLTranslationStrings prints comments and values as a list. This must be flattened for write_en_language.
+def prune_lists_to_strings(data):
+    for value in data.values():
+        if isinstance(value, dict):
+            for sub_key, sub_value in value.items():
+                if isinstance(sub_value, list) and sub_value:
+                    value[sub_key] = sub_value[0]
+        else:
+            print("Unexpected input")
+            sys.exit(1)
+    return data
+
+print("Preparing the addons shared string file")
+
+# Make sure we have all the required things
+# Lookup our required tools for addon generation.
+
+qtbinpath = qtquery('qmake', 'QT_INSTALL_BINS')
+if qtbinpath is None:
+    qtbinpath = qtquery('qmake6', 'QT_INSTALL_BINS')
+if qtbinpath is None:
+    print('Unable to locate qmake tool.')
```

Comment:
```suggestion
    sys.exit('Unable to locate qmake tool.')
```

You can exit directly with a message, same result a printing a message followed by `sys.exit(1)`.

Reply:
Thanks, updated